src/libraries/System.Private.CoreLib/src/System/Diagnostics/Tracing/EventSource.cs

```diff
@@ -2663,6 +2660,9 @@ internal void DoCommand(EventCommandEventArgs commandArgs)
                             m_eventSourceEnabled = false;
                         }
                     }
+
+                    this.OnEventCommand(commandArgs);
```

Comment:
The previous asymmetry in when the callback ran relative to when the IsEnabled() bit updated puzzled me and this certainly seems more intuitive. However its possible the previous behavior was designed to let people send events during the disable callback and now they would be unable to do so. You should probably test it. If we are lucky sending events during the Disable callback already didn't work in which case this re-ordering is probably safe.

Reply:
Verified that in the existing code you cannot send events during a Disable command, the event is already turned off